### ISSUE TITLE
[BUG] ForecastingPipeline cannot be constructed due to erroneous type check

### ISSUE THREAD

**Issue by fkiraly** [alan-turing-institute/sktime#2455]
The following `ForecastingPipeline` cannot be constructed due to what looks like an erroneous type check.

Expected behaviour is that it constructs without error.
Works on 0.11.0, but not on 0.11.2.

```python
from sktime.forecasting.compose import ForecastingPipeline, TransformedTargetForecaster
from sktime.transformations.series.compose import OptionalPassthrough
from sktime.transformations.series.boxcox import LogTransformer
from sktime.forecasting.fbprophet import Prophet
from sktime.transformations.series.detrend import Detrender
from sktime.forecasting.ets import AutoETS

pipe_prophet = ForecastingPipeline(steps=[
        ("logX", OptionalPassthrough(LogTransformer())),
        ("detrenderX", OptionalPassthrough(Detrender(forecaster = AutoETS()))),
        ("prophetforecaster", TransformedTargetForecaster(steps = [
            ("log", OptionalPassthrough(LogTransformer())),
            ("prophet", Prophet(yearly_seasonality='auto'))
        ]))  
])
```

**Comment by fkiraly** [comment]
The cause of this is that `TransformedTargetForecaster` has, in theory, *two* scitypes, per the inheritance check.

These are not correctly returned by the `scitype` utility which has a bug.
The check in ForecastingPipeline` also erroneously assumes that an estimator can have only one scitype.

**Comment by ngupta23** [comment]
FYI...I am getting this in pycaret as well after upgrading to 0.11.2

```python
from sktime.forecasting.naive import NaiveForecaster
from sktime.forecasting.compose import ForecastingPipeline, TransformedTargetForecaster

forecaster = TransformedTargetForecaster(
    steps=[
        ("model", NaiveForecaster(strategy="drift"))
    ]
)

pipe = ForecastingPipeline(
    steps=[("forecaster", forecaster)]
)
```

```python-traceback
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
[<ipython-input-15-5a138d05a5e2>](https://localhost:8080/#) in <module>()
     10 
     11 pipe = ForecastingPipeline(
---> 12     steps=[("forecaster", forecaster)]
     13 )
     14 # pipe.fit(y_train)

1 frames
[/usr/local/lib/python3.7/dist-packages/sktime/forecasting/compose/_pipeline.py](https://localhost:8080/#) in _check_steps(self, estimators, allow_postproc)
     62         if not set(scitypes).issubset(["forecaster", "transformer"]):
     63             raise TypeError(
---> 64                 f"estimators passed to {type(self).__name__} "
     65                 f"must be either transformer or forecaster"
     66             )

TypeError: estimators passed to ForecastingPipeline must be either transformer or forecaster
```

**Comment by fkiraly** [comment]
yes, will be fixed once #2456 is merged.
Sorry for this - this is resulting from the pipeline being both forecaster and transformer.
